keras/layers/normalization.py

```diff
@@ -188,6 +188,8 @@ def normalize_inference():
             sample_size = K.prod([K.shape(inputs)[axis]
                                   for axis in reduction_axes])
             sample_size = K.cast(sample_size, dtype=K.dtype(inputs))
+            if K.backend() == 'tensorflow' and sample_size.dtype != 'float32':
```

Comment:
`sample_size.dtype != 'float32'` is always going to be true, I think, because `sample_size.dtype` is a dtype object and not a string.

Reply:
No, I think it should work, for example if you try:
```python
a = keras.backend.arange(start=0, stop=10, step=1, dtype='int32')
a.dtype != 'int32'
```
returns False

Reply:
The lines where this is handled:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/framework/dtypes.py#L271-L272